src/test/java/io/cdap/plugin/facebook/source/common/config/BaseSourceConfigTest.java

```diff
@@ -0,0 +1,151 @@
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.facebook.source.common.config;
+
+import com.facebook.ads.sdk.AdsInsights;
+import io.cdap.cdap.etl.mock.validation.MockFailureCollector;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class BaseSourceConfigTest {
+  static class BaseSourceConfigMock extends BaseSourceConfig {
+
+    BaseSourceConfigMock() {
+      super("ref");
+    }
+
+    static BaseSourceConfig withIdsAndType(String adId, String adSetId, String accountId, String campaignId,
+                                           String objectType) {
+      BaseSourceConfig result = new BaseSourceConfigMock();
+      result.adId = adId;
+      result.adSetId = adSetId;
+      result.accountId = accountId;
+      result.campaignId = campaignId;
+      result.objectType = objectType;
+      return result;
+    }
+
+    static BaseSourceConfig withFiltering(String filtering) {
+      BaseSourceConfigMock result = new BaseSourceConfigMock();
+      result.filtering = filtering;
+      return result;
+    }
+
+    static BaseSourceConfig withFields(String fields) {
+      BaseSourceConfigMock result = new BaseSourceConfigMock();
+      result.fields = fields;
+      return result;
+    }
+
+    static BaseSourceConfig withBreakdown(String breakdown) {
+      BaseSourceConfigMock result = new BaseSourceConfigMock();
+      result.breakdown = breakdown;
+      return result;
+    }
+  }
+
+  @Test
+  public void testValidateObjectIdEmptyId() {
+    BaseSourceConfig config = BaseSourceConfigMock.withIdsAndType("", "", "", "", "Ad");
+    MockFailureCollector failureCollector = new MockFailureCollector();
+    config.validateObjectId(failureCollector);
+
+    Assert.assertEquals(1, failureCollector.getValidationFailures().size());
+  }
+
+  @Test
+  public void testValidateObjectId() {
+    BaseSourceConfig config = BaseSourceConfigMock.withIdsAndType("adId", "", "", "", "Ad");
+    MockFailureCollector failureCollector = new MockFailureCollector();
+    config.validateObjectId(failureCollector);
+
+    Assert.assertTrue(failureCollector.getValidationFailures().isEmpty());
+    Assert.assertEquals("adId", config.getObjectId());
+  }
+
+  @Test
+  public void testValidateFields() {
+    BaseSourceConfig config = BaseSourceConfigMock.withFields("date_start,impressions");
+    MockFailureCollector failureCollector = new MockFailureCollector();
+    config.validateFields(failureCollector);
+
+    Assert.assertTrue(failureCollector.getValidationFailures().isEmpty());
+  }
+
+  @Test
+  public void testValidateFieldsInvalidField() {
+    BaseSourceConfig config = BaseSourceConfigMock.withFields("date_start,impressions,invalid");
+    MockFailureCollector failureCollector = new MockFailureCollector();
+    config.validateFields(failureCollector);
+
+    Assert.assertEquals(1, failureCollector.getValidationFailures().size());
```

Comment:
can you use validation utils like in other configs? I don't think testing failures count is enough.